com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -267,7 +267,19 @@ public static ObservationProto GetObservationProto(this ISensor sensor, Observat
         {
             var shape = sensor.GetObservationShape();
             ObservationProto observationProto = null;
-            if (sensor.GetCompressionType() == SensorCompressionType.None)
+            var compressionType = sensor.GetCompressionType();
+            // Check capabilities if we need to concatenate PNGs
+            if (compressionType == SensorCompressionType.PNG && shape.Length == 3 && shape[2] > 3)
+            {
+                var trainerCanHandle = Academy.Instance.TrainerCapabilities == null || Academy.Instance.TrainerCapabilities.ConcatenatedPngObservations;
+                if (!trainerCanHandle)
+                {
+                    Debug.LogWarning("Attached trainer doesn't support multiple PNGs. Switching to uncompressed observations.");
```

Comment:
I fear this warning will be thrown a lot throughout the simulation. Should we set the compression type to None in the sensor?

Reply:
Good call. I'll make sure it only fires once.

> Should we set the compression type to None in the sensor?

We don't currently have an interface for setting the compression type. I didn't want to make the sensor implementations check the compatibility, since they might forget.